chevah/compat
# [#4874] Raise OSError on setting permission instead of AttributeError.

Scope
=====

This updated the Windows behaviour to look a bit closer to Unix when setting owner and group.


Changes
=======

Always raise OSError

As a drive by I created separeate tests for code which has different behaviour on Linux and Windows.

As a drive-by change I wrote a test for setAttribute for time uid and mode.


How to try and test the changes
===============================

reviewers: @dumol 

@dumol  Check to see if you can make sense of the code.

I would also like to use this PR to make sure that compat test are green in order to support the python-package.

I am also updating the openbsd part.
